build.sh

```diff
@@ -112,6 +112,15 @@ fi
 if [[ -z ${USE_SSE4_2} ]]; then
     USE_SSE4_2=ON
 fi
+# detect cpuinfo
+if [[ -z $(grep -o 'avx[^ ]*' /proc/cpuinfo) ]]; then
+    USE_AVX2=OFF
+fi
+
+if [[ -z $(grep -o 'sse[^ ]*' /proc/cpuinfo) ]]; then
```

Comment:
I have a question. If this is executed on the intel chip and macos, AVX2 and SSE will be disabled.
Could you make them available too?

Reply:
@luohaha Do you want to improve this on this PR, or I merge this and maybe work on this later?

Reply:
@imay On macos, `/proc` not supported. We can only support compile starrocks on linux now, so I don't think it's a problem.